library/ssl_tls13_client.c

```diff
@@ -2686,7 +2687,7 @@ static int ssl_encrypted_extensions_parse( mbedtls_ssl_context* ssl,
             case MBEDTLS_TLS_EXT_ALPN:
                 MBEDTLS_SSL_DEBUG_MSG( 3, ( "found alpn extension" ) );
 
-                if( ( ret = ssl_parse_alpn_ext( ssl, ext + 4, (size_t)ext_size ) ) != 0 )
+                if( ( ret = ssl_parse_alpn_ext( ssl, ext + 4, (size_t) ext_size ) ) != 0 )
```

Comment:
Overly long line

Reply:
Overly long line also in the MFL extension above.

Reply:
Wrong spacing in `switch( ext_id )` above.

Reply:
This is reusing the ALPN extension parser from `ssl_cli.c`, which assumes that the ALPN extension is part of the `ServerHello`. We need to modify the parser in such a way that we can use it from both the `ServerHello` and `EncryptedExtensions` parser.

One way to do this would be to return `0` (success) or non-zero (failure) and have the caller re-map this to the appropriate message-specific error code.

Another way would be to have generic error codes which don't distinguish between HS messages - as mentioned, I've raised this with the team.

Reply:
The same comment applies to the MFL parsing.